--- a/net/minecraft/client/renderer/block/model/VariantSelector.java
+++ b/net/minecraft/client/renderer/block/model/VariantSelector.java
@@ -26,7 +_,7 @@
                 Property<?> property = stateDefinition.getProperty(string2);
                 if (property != null && iterator.hasNext()) {
                     String string3 = iterator.next();
-                    Comparable<?> valueHelper = getValueHelper((Property<Comparable<?>>)property, string3);
+                    Comparable<?> valueHelper = getValueHelper(property, string3);
                     if (valueHelper == null) {
                         throw new RuntimeException(
                             "Unknown value: '" + string3 + "' for blockstate property: '" + string2 + "' " + property.getPossibleValues()
