--- a/net/minecraft/client/renderer/LevelEventHandler.java
+++ b/net/minecraft/client/renderer/LevelEventHandler.java
@@ -609,9 +_,9 @@
                 int i5 = i1 >> 6;
                 if (i5 > 0) {
                     if (randomSource.nextFloat() < 0.3F + i5 * 0.1F) {
-                        float f2 = 0.15F + 0.02F * i5 * i5 * randomSource.nextFloat();
-                        float f4 = 0.4F + 0.3F * i5 * randomSource.nextFloat();
-                        this.level.playLocalSound(blockPos, SoundEvents.SCULK_BLOCK_CHARGE, SoundSource.BLOCKS, f2, f4, false);
+                        float volume = 0.15F + 0.02F * i5 * i5 * randomSource.nextFloat();
+                        float pitch = 0.4F + 0.3F * i5 * randomSource.nextFloat();
+                        this.level.playLocalSound(blockPos, SoundEvents.SCULK_BLOCK_CHARGE, SoundSource.BLOCKS, volume, pitch, false);
                     }
 
                     byte b = (byte)(i1 & 63);
