--- a/net/minecraft/client/gui/screens/PresetFlatWorldScreen.java
+++ b/net/minecraft/client/gui/screens/PresetFlatWorldScreen.java
@@ -24,6 +_,7 @@
 import net.minecraft.core.registries.Registries;
 import net.minecraft.network.chat.CommonComponents;
 import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.MutableComponent;
 import net.minecraft.resources.ResourceKey;
 import net.minecraft.resources.ResourceLocation;
 import net.minecraft.tags.FlatLevelGeneratorPresetTags;
@@ -299,7 +_,7 @@
                 this.preset = presetHolder.value();
                 this.name = presetHolder.unwrapKey()
                     .map(resourceKey -> Component.translatable(resourceKey.location().toLanguageKey("flat_world_preset")))
-                    .orElse(PresetFlatWorldScreen.UNKNOWN_PRESET);
+                    .orElse((MutableComponent) PresetFlatWorldScreen.UNKNOWN_PRESET);
             }
 
             @Override
