--- a/net/minecraft/client/gui/screens/inventory/BookViewScreen.java
+++ b/net/minecraft/client/gui/screens/inventory/BookViewScreen.java
@@ -10,11 +_,7 @@
 import net.minecraft.client.gui.screens.Screen;
 import net.minecraft.client.renderer.RenderType;
 import net.minecraft.core.component.DataComponents;
-import net.minecraft.network.chat.ClickEvent;
-import net.minecraft.network.chat.CommonComponents;
-import net.minecraft.network.chat.Component;
-import net.minecraft.network.chat.FormattedText;
-import net.minecraft.network.chat.Style;
+import net.minecraft.network.chat.*;
 import net.minecraft.resources.ResourceLocation;
 import net.minecraft.util.FormattedCharSequence;
 import net.minecraft.util.Mth;
@@ -258,7 +_,10 @@
             } else {
                 WritableBookContent writableBookContent = stack.get(DataComponents.WRITABLE_BOOK_CONTENT);
                 return writableBookContent != null
-                    ? new BookViewScreen.BookAccess(writableBookContent.getPages(isTextFilteringEnabled).map(Component::literal).toList())
+                    ? new BookViewScreen.BookAccess(writableBookContent
+                            .getPages(isTextFilteringEnabled)
+                            .map((page) -> (Component)Component.literal(page)
+                        ).toList())
                     : null;
             }
         }
